api/v1alpha1/params_validation.go

```diff
@@ -0,0 +1,151 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"github.com/azure/kaito/pkg/k8sclient"
+	"github.com/azure/kaito/pkg/utils"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"knative.dev/pkg/apis"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type TrainingConfig struct {
+	ModelConfig        map[string]interface{} `yaml:"ModelConfig"`
+	TokenizerParams    map[string]interface{} `yaml:"TokenizerParams"`
+	QuantizationConfig map[string]interface{} `yaml:"QuantizationConfig"`
+	LoraConfig         map[string]interface{} `yaml:"LoraConfig"`
+	TrainingArguments  map[string]interface{} `yaml:"TrainingArguments"`
+	DatasetConfig      map[string]interface{} `yaml:"DatasetConfig"`
+	DataCollator       map[string]interface{} `yaml:"DataCollator"`
+}
+
+func validateMethodViaConfigMap(cm *corev1.ConfigMap, methodLowerCase string) *apis.FieldError {
+	trainingConfigYAML, ok := cm.Data["training_config.yaml"]
+	if !ok {
+		return apis.ErrGeneric(fmt.Sprintf("ConfigMap '%s' does not contain 'training_config.yaml' in namespace '%s'", cm.Name, cm.Namespace), "config")
+	}
+	fmt.Printf("1Failed validateMethodViaConfigMap")
+
+	var trainingConfig TrainingConfig
+	if err := yaml.Unmarshal([]byte(trainingConfigYAML), &trainingConfig); err != nil {
+		return apis.ErrGeneric(fmt.Sprintf("Failed to parse 'training_config.yaml' in ConfigMap '%s' in namespace '%s': %v", cm.Name, cm.Namespace, err), "config")
+	}
+	fmt.Printf("1Failed validateMethodViaConfigMap")
+
+	// Validate QuantizationConfig if it exists
+	quantConfig := trainingConfig.QuantizationConfig
+	if quantConfig != nil {
+		// Dynamic field search for quantization settings within ModelConfig
+		loadIn4bit, _ := utils.SearchMap(quantConfig, "load_in_4bit")
+		loadIn8bit, _ := utils.SearchMap(quantConfig, "load_in_8bit")
+
+		loadIn4bitBool, ok4bitBool := loadIn4bit.(bool)
```

Comment:
You need to do a null check for loadIn4bit and loadIn8bit. 

I don't quite understand the logic here. loadIn4bit should be a string "true" or "false" , right? How to convert a string to a bool? or the yaml deserialization will turn ture/false to 1/0?

Reply:
I think if !ok4bitbool, you can fail the check to simplify the follow up code.

Reply:
There are 3 different combinations of results from the casting. 

`loadIn4bitBool, ok4bitBool := loadIn4bit.(bool)`

```
in the case loadIn4bit is nil
loadIn4bitBool is false, and ok4bitBool is false

in the case loadIn4bit is false
loadIn4bitBool is false, and ok4bitBool is true

in the case loadIn4bit is true
loadIn4bitBool is true, and ok4bitBool is true
```